### PR TITLE
Support vesting a partial unpaid amount

### PR DESCRIPTION
**Pull request by talbaneth**
Supports vesting only a part of the unpaid amount through a new _maxAmt parameter to vest().

### CODE REVIEW

File: src/DssVest.t.sol [hunk 1]
```diff
@@ -489,4 +490,28 @@ contract DssVestTest is DSTest {
...
+
+        vest.vest(id, 5 * days_vest);
+
+        (usr, bgn, clf, fin, amt, rxd, mgr)  = vest.awards(id);
+        assertEq(usr, address(this));
+        assertEq(uint256(bgn), now - 10 days);
+        assertEq(uint256(fin), now + 90 days);
+        assertEq(uint256(amt), 100 * days_vest);
+        assertEq(uint256(rxd), 5 * days_vest);
+        assertEq(Token(address(vest.gem())).balanceOf(address(this)), 5 * days_vest);
```
Comment (hexonaut) [inline src/DssVest.t.sol:515]: I think we could maybe expand this test a bit to include multiple calls to `vest` to show that you can both take up to the amount owed at this time (10 days worth), and that it correctly works with subsequent elapsed time.

**Review by hexonaut — COMMENTED** [review]
Code looks good. I think the test should be expanded a bit, but other than that looks good to merge.

File: src/DssVest.t.sol (hunk 1 above)
Comment (talbaneth) [reply src/DssVest.t.sol]: Done

**Review by hexonaut — APPROVED** [review]
LGTM. I think we should include this.

**Review by kmbarry1 — APPROVED** [review]
LGTM

**Comment by gbalabasquer** [comment]
> Others aren't typically a fan of this, but I would enjoy an overloaded function that still allows a user to just pass their token ID.
> 
> ```
> function vest(uint256 id) external {
>     vest(id, uint256(-1));
> }
> ```
> 
> This contract will be used directly by non-engineer users, and through multi-sig contracts. We should make it as simple to interact with as possible.

Let's do this change and I'd suggest rolling back all the other tests that won't use the partial function so we keep the final changes of this PR the minimum as possible.

**Comment by brianmcmichael** [comment]
PR against tal's branch
https://github.com/talbaneth/dss-vest/pull/1

**Pull request merged**